src/game_server/player_update_packet.rs

```diff
@@ -21,6 +36,215 @@ impl SerializePacket for PlayerUpdateOpCode {
+}
+
+#[derive(SerializePacket, DeserializePacket)]
+pub struct HudMessage {
+	guid: u64,
+	unkguid: u64,
+	unknown1: u32,
+	unknown2: u32,
+	unknown3: u32,
+	unknown4: u32,
+	unknown5: u32,
+	unknown6: u32,
+}
+
+impl GamePacket for HudMessage {
+	type Header = PlayerUpdateOpCode;
+	const HEADER: Self::Header = PlayerUpdateOpCode::HudMessage;
+}
+
+#[derive(SerializePacket, DeserializePacket)]
+pub struct SlotCompositeEffectOverride {
+	guid: u64,
+	slot_id: u32,
+	composite_effect: u32,
+}
+
+impl GamePacket for SlotCompositeEffectOverride {
+	type Header = PlayerUpdateOpCode;
+	const HEADER: Self::Header = PlayerUpdateOpCode::SlotCompositeEffectOverride;
+}
+
+#[derive(SerializePacket, DeserializePacket)]
+pub struct UpdateRemoveTemporaryAppearance {
+	guid: u64,
+	model_id: u32,
+}
+
+impl GamePacket for UpdateRemoveTemporaryAppearance {
+	type Header = PlayerUpdateOpCode;
+	const HEADER: Self::Header = PlayerUpdateOpCode::UpdateRemoveTemporaryAppearance;
+}
+
+#[derive(SerializePacket, DeserializePacket)]
+pub struct UpdateTemporaryAppearance {
+	model_id: u32,
+	guid: u64,
+}
+
+impl GamePacket for UpdateTemporaryAppearance {
+	type Header = PlayerUpdateOpCode;
+	const HEADER: Self::Header = PlayerUpdateOpCode::UpdateTemporaryAppearance;
+}
+
+#[derive(SerializePacket, DeserializePacket)]
+pub struct SetSpawnerActivationEffect {
+	guid: u64,
+	composite_effect: u32,
+}
+
+impl GamePacket for SetSpawnerActivationEffect {
+	type Header = PlayerUpdateOpCode;
+	const HEADER: Self::Header = PlayerUpdateOpCode::SetSpawnerActivationEffect;
+}
+
+[derive(SerializePacket, DeserializePacket)]
+pub struct MoveOnRelativeRail {
+	pub guid: u64,
+	pub unknown1: u32,
+	pub unknown2: u32,
+	pub unknown3: u32,
+	pub unknown4: u32,
+	pub unknown5: u32,
+	pub unknown6: Pos,
+}
+
+impl GamePacket for MoveOnRelativeRail {
+	type Header = PlayerUpdateOpCode;
+	const HEADER: Self::Header = PlayerUpdateOpCode::MoveOnRelativeRail;
+}
+
+#[derive(SerializePacket, DeserializePacket)]
+pub struct ClearRail {
+	pub guid: u64,
+	
+}
+
+impl GamePacket for ClearRail {
+	type Header = PlayerUpdateOpCode;
+	const HEADER: Self::Header = PlayerUpdateOpCode::ClearRail;
+}
+
+#[derive(SerializePacket, DeserializePacket)]
+pub struct MoveOnRail {
+	 guid: u64,
+	 unknown1: u32,
+	 unknown2: u32,
+	 position: Pos,
+	
+}
+
+impl GamePacket for MoveOnRail {
+	type Header = PlayerUpdateOpCode;
+	const HEADER: Self::Header = PlayerUpdateOpCode::MoveOnRail;
+}
+
+#[derive(SerializePacket, DeserializePacket)]
+pub struct SeekTargetUpdate {
+	pub guid: u64,
+	pub targetid: u64,
+}
+
+impl GamePacket for SeekTargetUpdate {
+	type Header = PlayerUpdateOpCode;
+	const HEADER: Self::Header = PlayerUpdateOpCode::SeekTargetUpdate;
+}
+
+#[derive(SerializePacket, DeserializePacket)]
+pub struct SeekTarget {
+	pub guid: u64,
+	pub targetid: u64,
+	pub initspeed: f32,
+	pub acceleration: f32,
+	pub speed: f32,
+	pub unknown1: f32,
+	pub yrot: f32,
+	pub rotation: Pos,
```

Comment:
```suggestion
	pub rot_y: f32,
	pub rot: Pos,
```
Also, it's fitting that they created a y rotation field followed by a float array for rotation, lol